pfinance/functions.py

```diff
@@ -257,6 +257,49 @@ def straight_line_depreciation(purchase_price: float, salvage_value: float, usef
     return (purchase_price - salvage_value) / useful_life
 
 
+def double_declining_balance_depreciation(
+    purchase_price: float,
+    salvage_value: float,
+    useful_life: int,
+    factor: float = 2.0,
+) -> dict[list[float], list[float]]:
```

Comment:
I believe this should be `dict[string, list[float]]` (annotation shows the key type and value type)